pkgs/build-support/testers/expect-failure.sh

```diff
@@ -35,7 +35,28 @@ echo "testBuildFailure: Original builder produced exit code: $r"
 # -----------------------------------------
 # Write the build log to the default output
 #
-# # from stdenv setup.sh
+# # based on stdenv setup.sh
+if [ -f .attrs.sh ] || [[ -n "${NIX_ATTRS_SH_FILE:-}" ]] || [[ -n "${NIX_ATTRS_JSON_FILE:-}" ]]; then
```

Comment:
```suggestion
if [ -f .attrs.sh ] || [[ -n "${NIX_ATTRS_SH_FILE:-}" ]]; then
```

I think by the time we get here, `NIX_ATTRS_JSON_FILE` has already been coerced into a `NIX_ATTRS_SH_FILE`?

---

In fact, I think this can actually be:
```suggestion
if [[ -n "${NIX_ATTRS_SH_FILE:-}" ]]; then
```
Since I believe that var is reliably defined in this context?

---

Not sure how best to go about testing this. Do you know of any derivations that'd run into edge-cases here?